ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/json/ChainIdDeserializer.java

```diff
@@ -35,12 +35,7 @@ public ChainIdDeserializer(final Class<?> vc) {
   @Override
   public BigInteger deserialize(final JsonParser jsonparser, final DeserializationContext context)
       throws IOException {
-    final var chainId =
-        UInt256.fromHexString(jsonparser.getCodec().readValue(jsonparser, String.class))
-            .toBigInteger();
-    if (chainId.signum() <= 0) {
-      throw new IllegalArgumentException("Non positive chain id: " + chainId);
-    }
```

Comment:
@fab-10 FYI EIP-7702 formalises `chainId = 0` being allowed
See https://eips.ethereum.org/EIPS/eip-7702 and
https://ethereum-magicians.org/t/eip-2294-explicit-bound-to-chain-id/11090/29?u=siladu

Reply:
Is chainId = 0 allowed everywhere? When I read 7702 I thought it only applied when processing the new transaction type.

Reply:
@jframe I'm not sure. Do you know what impact this could have in other non-7702 scenarios?

Hoping @fab-10 will advise on why this check was added in originally.

I can create a ref test specific deserializer as a precaution but would rather not add more code than necessary.

Reply:
I see no harm in allowing the `chainId` to be zero everywhere, but instead of completely remove the validation, you can change it to only check it is not negative.

Reply:
Good point @fab-10 , added that back in